File: Task/Narcissistic-decimal-number/Go/narcissistic-decimal-number.go
package main

import "fmt"

func narc(n int) []int {
	power := [...]int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
	limit := 10
	result := make([]int, 0, n)
	for x := 0; len(result) < n; x++ {
		if x >= limit {
			for i := range power {
				power[i] *= i // i^m
			}
			limit *= 10
		}
		sum := 0
		for xx := x; xx > 0; xx /= 10 {
			sum += power[xx%10]
		}
		if sum == x {
			result = append(result, x)
		}
	}
	return result
}

func main() {
	fmt.Println(narc(25))
}

File: Task/Narcissistic-decimal-number/Zen-C/narcissistic-decimal-number.c
import "std/vec.zc"

fn narc(n: int) -> Vec<int> {
    let power: [int; 10];
    for i in 0..10 { power[i] = i; }
    let limit = 10;
    let res = Vec<int>::new();
    for let x = 0; res.length() < n; ++x {
        if x >= limit {
            for i in 0..10 { power[i] *= i; }
            limit *= 10;
        }
        let sum = 0;
        let xx = x;
        while xx > 0 {
            sum += power[xx % 10];
            xx /= 10;
        }
        if sum == x { res << x; }
    }
    return res;
}

fn main() {
    println "The first 25 narcissistic decimal numbers are:"
    let res = narc(25);
    for n in res {
        print "{n} ";
    }
    println "";
}
